_posts/en/newsletters/2019-07-31-newsletter.md

```diff
@@ -0,0 +1,241 @@
+    amounts of bitcoins or the ability to sell bitcoins, making sybil
+    attacks more expensive.  On the downside, honest makers would likely
+    charge more for coinjoins in order to receive compensation for their
+    sacrifices, raising the cost of JoinMarket coinjoins.
+
+- **PR opened for BIP322 generic signed message format:** Kalle Alm
+  [opened a PR][Bitcoin Core #16440] to Bitcoin Core with an
+  implementation of [BIP322][] that updates the `signmessage`
+  and `verifymessage` RPCs to support signing of arbitrary scripts
+  rather than just P2PKH templates.  The `signmessagewithprivkey` RPC (sign
+  message with private key) is also updated to allow signing with
+  P2WPKH or P2SH-P2WSH addresses in addition to legacy P2PKH.  The PR is currently
+  seeking feedback and conceptual review.  If merged, BIP322 support
+  will allow Bitcoin Core to sign and verify messages for
+  any script, with such support easily upgradable for future
+  changes in the scripting language such as [taproot][bip-taproot].
+  Hopefully other wallets would then consider implementing BIP322
+  themselves to provide flexible forward-compatible message signing
+  support.  See Optech's previous [bech32 section][signmessage bech32]
+  where we lament the previous lack of a BIP322 implementation.
+
+    Based on early feedback on the PR, BIP322 has also been updated so
+    that signing and verifying messages with legacy P2PKH keys uses the
+    old signmessage format.  This allows BIP322 tools to be fully backwards
+    compatible with existing and widely-implemented signmessage standard
+    (which only supports P2PKH).
+
+- **Bloom filter discussion:** in the *notable changes* section of [last
+  week's newsletter][Newsletter #56], we mentioned a [merged PR][Bitcoin
+  Core #16152] that disabled bloom filters in Bitcoin Core's default
+  configuration.  The author of the PR [announced][bloom announce] this
+  unreleased change to the Bitcoin-Dev mailing list and several people
+  replied with questions or concerns.  Some takeaways from the
+  discussion include:
+
+    - *No urgent action required:* as this writing, over 20% of nodes
+      accepting incoming connections (listening nodes) are running a
+      version of Bitcoin Core that's over a year old [according to
+      BitNodes][bitnodes dashboard].  At least 5% are running a version
+      that's over two years old.  Extrapolating to the future, that
+      means there will still be over 500 listening nodes providing
+      bloom filters at the beginning of 2022.  So, even with no further
+      action, wallet developers may have a significant amount of time in
+      which to adapt their programs.
+
+    - *Spies likely to run their own nodes:* additional to non-upgraded
+      nodes, it's likely spy nodes will continue to provide bloom filter
+      support in the future given that the way most wallets use bloom
+      filters creates a severe privacy leak.
```

Comment:
Perhaps another way to say this: "There is a financial incentive for chain analysis companies to provide as much possible full node capacity as there is BIP37 client demand given the privacy leaks.